packages/cli/src/evaluation/metrics.controller.ts

```diff
@@ -0,0 +1,127 @@
+import express from 'express';
+
+import { TestMetricRepository } from '@/databases/repositories/test-metric.repository.ee';
+import { Delete, Get, Patch, Post, RestController } from '@/decorators';
+import { NotFoundError } from '@/errors/response-errors/not-found.error';
+import { testMetricCreateRequestBodySchema } from '@/evaluation/metric.schema';
+import { getSharedWorkflowIds } from '@/public-api/v1/handlers/workflows/workflows.service';
+
+import { TestDefinitionService } from './test-definition.service.ee';
+import { TestMetricsRequest } from './test-definitions.types.ee';
+
+@RestController('/evaluation/test-definitions')
+export class TestMetricsController {
+	constructor(
+		private readonly testDefinitionService: TestDefinitionService,
+		private readonly testMetricRepository: TestMetricRepository,
+	) {}
+
+	// This method is used in multiple places in the controller to get the test definition
+	// (or just check that it exists and the user has access to it).
+	private async getTestDefinition(
+		req:
+			| TestMetricsRequest.GetOne
+			| TestMetricsRequest.GetMany
+			| TestMetricsRequest.Patch
+			| TestMetricsRequest.Delete
+			| TestMetricsRequest.Create,
+	) {
+		const { testDefinitionId } = req.params;
```

Comment:
Should we check that `testDefinitionId` is not empty?

Reply:
Should be fine without check. It's a path parameter, so in case it's not present – the routes will not match in a first place.